src/pgwire/src/protocol.rs

```diff
@@ -1017,8 +1053,9 @@ where
         let message: BackendMessage = message.into();
         match message {
             BackendMessage::ErrorResponse(ref err) => {
+                // TODO(jkosh44) this is wrong. We may have lost the client at this point.
                 let minimum_client_severity =
-                    self.adapter_client.session().vars().client_min_messages();
+                    self.adapter_client().session().vars().client_min_messages();
```

Comment:
Ok I fixed this in this commit:https://github.com/MaterializeInc/materialize/pull/15873/commits/2ca2d0942a189ff231cd6a8ac3b38331c32d41c4. You may want to take a look at that since there's a decent amount of changes.